playwright/tests/e2e/steps/transactions-overview.steps.ts

```diff
@@ -0,0 +1,221 @@
+
+import { transactionOverviewPage } from "../../../pages/transaction-overview.page";
+import { transactionDetailPage } from "../../../pages/transaction-detail.page";
+
+const { Given, When, Then } = createBdd();
+
+let addressId: string | null;
+let transactionHash: string | null;
+
+Given(/^the user is in the general dashboard page in explorer portal test$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToDashboard();
+});
+When(/^the user selects the Transactions option inside the Blockchain drop down menu$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToEpochsFromSidebar();
+});
+Then(
+  /^the user should see the transactions page containing the Search bar and Transactions table$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).searchBarOnTransaction();
+    await transactionOverviewPage(page).checkTransactionsTable();
+  }
+);
+
+Given(/^the user is in the transactions page in explorer portal site$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects any data of the row execpt the transaction hash, block number, epoch number or addresses hash of one of the transactions record in the table$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).openWidget();
+  }
+);
+Then(
+  /^the user should see the Info widget data with the respective data of the selected transaction$/,
+  async ({ page, request }) => {
+    const trxHash = await page.getByTestId("transaction.table.value.txhash#0");
+    const ariaLabelValue = await trxHash?.getAttribute("aria-label");
+    if (ariaLabelValue) {
+      (await koiosApi(request).getTransactionByTrxHash(ariaLabelValue)).json().then(async (data) => {
+        await transactionOverviewPage(page).checkCurrenTransactionWidget({ currentTransaction: data });
+      });
+    }
+  }
+);
+
+Given(/^the user is in the Transactions page$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the transaction hash of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.txhash#0");
+    transactionHash = await transactionsTableValueEpoch.getAttribute("href");
+    await transactionsTableValueEpoch.click();
+  }
+);
+Then(/^the transaction details page of the select transaction should be opened$/, async ({ page }) => {
+  await transactionDetailPage(page).checkTransactionsDetail({ transactionHash: transactionHash || "" });
+});
+
+Given(/^the user is in the Transactions page portal$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the block number of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.block#0");
```

Comment:
Same as above comment, let's keep locators interactions inside the pages functions and returns the values that you could need in the next steps if need it.